crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/MagsProceedingService.java

```diff
@@ -0,0 +1,81 @@
+            CaseType.INDICTABLE, CaseType.EITHER_WAY, CaseType.SUMMARY_ONLY
+    );
+
+    public MagsDecisionResult determineMagsRepDecision(CrownCourtDTO dto) {
+        ApiIOJSummary iojSummary = dto.getIojSummary();
+        ReviewResult iojResult = ReviewResult.getFrom(
+                iojSummary.getDecisionResult() != null
+                        ? iojSummary.getDecisionResult() : iojSummary.getIojResult()
+        );
+        boolean isMagsCourt = magsCourtCaseTypes.contains(dto.getCaseType());
+        if (iojResult != null && isMagsCourt) {
+            DecisionReason decisionReason = getDecisionReason(dto, iojResult);
+            if (decisionReason != null) {
+                MagsDecisionResult decisionResult =
+                        MagsDecisionResult.builder()
+                                .decisionDate(LocalDate.now())
+                                .decisionReason(decisionReason)
+                                .build();
+                dto.setMagsDecisionResult(decisionResult);
+                RepOrderDTO repOrderDTO =
+                        maatCourtDataService.updateRepOrder(UpdateRepOrderDTOBuilder.build(dto, decisionResult));
+                decisionResult.setTimestamp(repOrderDTO.getDateModified());
+                return decisionResult;
+            }
+        }
+        return null;
+    }
+
+    private DecisionReason getDecisionReason(CrownCourtDTO dto, ReviewResult iojResult) {
+        boolean isIojPassed = (iojResult == ReviewResult.PASS);
+        PassportAssessmentResult passportResult =
+                PassportAssessmentResult.getFrom(dto.getPassportAssessment().getResult());
+        boolean isPassported = passportResult != null &&
+                List.of(PassportAssessmentResult.PASS, PassportAssessmentResult.TEMP)
+                        .contains(passportResult);
+
+        ApiFinancialAssessment financialAssessment = dto.getFinancialAssessment();
+        InitAssessmentResult initResult = InitAssessmentResult.getFrom(financialAssessment.getInitResult());
+        FullAssessmentResult fullResult = FullAssessmentResult.getFrom(financialAssessment.getFullResult());
+        ReviewResult hardshipResult = ofNullable(financialAssessment.getHardshipOverview())
+                .map(ApiHardshipOverview::getReviewResult)
+                .orElse(null);
+
+        if (isPassported || initResult == InitAssessmentResult.PASS || hardshipResult == ReviewResult.PASS
+                || (initResult == InitAssessmentResult.FULL && fullResult == FullAssessmentResult.PASS)) {
+            return isIojPassed ? DecisionReason.GRANTED : DecisionReason.FAILIOJ;
+        } else if (initResult == InitAssessmentResult.FAIL || passportResult == PassportAssessmentResult.FAIL
+                || ((hardshipResult == null || hardshipResult == ReviewResult.FAIL) && fullResult == FullAssessmentResult.FAIL)) {
+            return isIojPassed ? DecisionReason.FAILMEANS : DecisionReason.FAILIOJ;
```

Comment:
When both IOJ and Means are fail, decision reason should be FAILMEIOJ (not FAILIOJ).